final pass over paper (#690)
JuliaSmoothOptimizers/Krylov.jl: paper/paper.md

The in-place variants only require a Julia structure that contains all the storage needed by a Krylov method as additional argument.
In-place methods limit memory allocations and deallocations, which are particularly expensive on GPUs.

## Performance optimizations and storage requirements

Operator-vector products and vector operations are the most expensive operations in Krylov.jl.
We rely on BLAS routines as much as possible to perform those operations.
By default, Julia ships with OpenBLAS and provides multithreaded routines.
Since Julia 1.6, users can also switch dynamically to other BLAS backends, such as the Intel MKL or BLIS, thanks to the BLAS demuxing library `libblastrampoline`, if an optimized BLAS is available.

A “Storage Requirements” section is available in the documentation to provide the theoretical number of bytes required by each method.
Our implementations are storage-optimal in the sense that they are guaranteed to match the theoretical storage amount.
The match is verified in the unit tests by way of functions that return the number of bytes allocated by our implementations.

# Examples

> Our first example is a simple implementation of Gauss-Newton method without linesearch for nonlinear least squares.

Reviewer: ```suggestion
Our first example is a simple implementation of the Gauss-Newton method without linesearch for nonlinear least squares.
```

It illustrates several of the facilities of Krylov.jl: solver preallocation and reuse, genericity with respect to data types, and linear operators.

> Another example based on a simplistic Newton's method without linesearch for convex optimization is also available in the documentation and illustrates the concepts separately in the sections “In-places methods” and “Factorization-free operators”.

Reviewer: ```suggestion
Another example based on a simplistic Newton method without linesearch for convex optimization is also available in the documentation, and illustrates the same concepts in the sections “In-places methods” and “Factorization-free operators”.
```

gauss_newton(F, JF, x₀)
```

> Our second example concerns the solution of a complex Hermitian linear system from the SuiteSparse Matrix Collection with an incomplete Cholesky preconditioner on GPU.

Reviewer: donner une référence.

> Our second example concerns the solution of a complex Hermitian linear system from the SuiteSparse Matrix Collection with an incomplete Cholesky preconditioner on GPU.

Reviewer: ```suggestion
Our second example concerns the solution of a complex Hermitian linear system from the SuiteSparse Matrix Collection with an incomplete Cholesky factorization preconditioner on GPU.
```

end

# Linear operator that model the preconditioner P⁻¹
T = ComplexF64
symmetric = false
hermitian = true
P⁻¹ = LinearOperator(T, m, n, symmetric, hermitian, (y, x) -> ldiv_ic0!(y, P, x))

> # Solve an Hermitian positive definite system with an incomplete Cholesky preconditioner on GPU

Reviewer: ```suggestion
# Solve a Hermitian positive definite system with an incomplete Cholesky factorization preconditioner on GPU
```
